frontend/src/pages/meditation/components/meditation-timer/meditation-timer.tsx

```diff
@@ -0,0 +1,80 @@
+import { Button } from '#libs/components/components.js';
+import {
+  useAppForm,
+  useCallback,
+  useFormController,
+} from '#libs/hooks/hooks.js';
+
+import { TimerButton } from './components/timer-button/timer-button.js';
+import { DURATION_UNIT, MEDITATION_DURATION } from './libs/constants.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  defaultDuration: string;
+  onStartSession: (duration: string) => void;
+};
+
+const MeditationTimer: React.FC<Properties> = ({
+  defaultDuration,
+  onStartSession,
+}) => {
+  const { control } = useAppForm({
+    defaultValues: { meditationDuration: defaultDuration },
+  });
+
+  const {
+    field: { onChange, value },
+  } = useFormController({
+    name: 'meditationDuration',
+    control,
+  });
+
+  const handleStartSession = useCallback(() => {
+    onStartSession(value);
+  }, [onStartSession, value]);
+
+  const startButtonText =
+    value === defaultDuration
+      ? 'Start with default duration'
+      : 'Start with selected duration';
+
+  return (
+    <div className={styles['timer']}>
+      <p className={styles['title']}>Choose Your Duration</p>
+      <div className={styles['duration-container']}>
+        {Object.keys(MEDITATION_DURATION).map((duration) => {
+          return (
+            <TimerButton
+              key={duration}
+              isActive={value === duration}
+              onChange={onChange}
+              value={duration}
+              name="Meditation Duration"
+              duration={
+                MEDITATION_DURATION[
+                  duration as keyof typeof MEDITATION_DURATION
```

Comment:
That may be redundant in case of improving const above

Reply:
Good point, thank you!

Reply:
@Dmytro-Kucherenko I did try to apply your advise but I wasn't able to simple remove `as keyof typeof MEDITATION_DURATION` because TS was treating `duration` as a generic string and not keys for duration const. 
I had to create a separate type for this, it kind of works but it feels like there is no unnecessarily more code. Could you have a look please?